<?xml version="1.0" ?>
<sdf version="1.7">
  <model name='freenove_4wd'>
    <pose>0 0 0 0 0 0</pose>

    <link name='chassis'>
      <inertial>
        <mass></mass>
        <inertia>
          <ixx>0.01</ixx>
          <iyy>0.01</iyy>
          <izz>0.01</izz>
        </inertia>
      </inertial>

      <visual name='chassis_visual'>
        <pose>0 0 0 0 0 0</pose>
        <geometry>
          <box>
            <size>0.2 0.1 0.03</size>
          </box>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/DarkGrey</name>
          </script>
        </material>
      </visual>

      <collision name='chassis_collision'>
        <pose>0 0 0 0 0 0</pose>
        <geometry>
          <box>
            <size>0.2 0.1 0.03</size>
          </box>
        </geometry>
      </collision>

      <visual name='computer_visual'>
        <pose>-0.05 0 0.03 0 0 0</pose>
        <geometry>
          <box>
            <size>0.1 0.07 0.01</size>
          </box>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/Green</name>
          </script>
        </material>
      </visual>

      <collision name='computer_collision'>
        <pose>-0.05 0 0.03 0 0 0</pose>
        <geometry>
          <box>
            <size>0.1 0.07 0.01</size>
          </box>
        </geometry>
      </collision>

    </link>
    <link name="front_left">
      <pose>0.08 0.07 -0.01 1.5707 0 0</pose>
      <inertial>
        <mass>0.01</mass>
        <inertia>
          <ixx>0.0001</ixx>
          <iyy>0.0001</iyy>
          <izz>0.0001</izz>
        </inertia>
      </inertial>

      <collision name="collision">
        <geometry>
          <cylinder>
            <radius>0.03</radius>
            <length>0.02</length>
          </cylinder>
        </geometry>
      </collision>

      <visual name="tire">
        <geometry>
          <cylinder>
            <radius>0.03</radius>
            <length>0.02</length>
          </cylinder>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/DarkGrey</name>
          </script>
        </material>
      </visual>

      <visual name="wheel">
        <geometry>
          <cylinder>
            <radius>0.028</radius>
            <length>0.022</length>
          </cylinder>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/Yellow</name>
          </script>
        </material>
      </visual>

      <visual name="wheel_line">
        <geometry>
          <box>
            <size>0.053 0.01 0.024</size>
          </box>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/DarkGrey</name>
          </script>
        </material>
      </visual>

      <visual name="wheel_line2">
        <geometry>
          <box>
            <size>0.01 0.053 0.024</size>
          </box>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/DarkGrey</name>
          </script>
        </material>
      </visual>

    </link>

    <joint type="revolute" name="joint_front_left">
      <child>front_left</child>
      <parent>chassis</parent>
      <axis>
        <xyz>0 0 1</xyz>
        <dynamics>
          <friction>0.01</friction>
          <damping>0.01</damping>
        </dynamics>
      </axis>
    </joint>
    
    <link name="front_right">
      <pose>0.08 -0.07 -0.01 1.5707 0 0</pose>
      <inertial>
        <mass>0.01</mass>
        <inertia>
          <ixx>0.0001</ixx>
          <iyy>0.0001</iyy>
          <izz>0.0001</izz>
        </inertia>
      </inertial>

      <collision name="collision">
        <geometry>
          <cylinder>
            <radius>0.03</radius>
            <length>0.02</length>
          </cylinder>
        </geometry>
      </collision>

      <visual name="tire">
        <geometry>
          <cylinder>
            <radius>0.03</radius>
            <length>0.02</length>
          </cylinder>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/DarkGrey</name>
          </script>
        </material>
      </visual>

      <visual name="wheel">
        <geometry>
          <cylinder>
            <radius>0.028</radius>
            <length>0.022</length>
          </cylinder>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/Yellow</name>
          </script>
        </material>
      </visual>

      <visual name="wheel_line">
        <geometry>
          <box>
            <size>0.053 0.01 0.024</size>
          </box>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/DarkGrey</name>
          </script>
        </material>
      </visual>

      <visual name="wheel_line2">
        <geometry>
          <box>
            <size>0.01 0.053 0.024</size>
          </box>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/DarkGrey</name>
          </script>
        </material>
      </visual>

    </link>

    <joint type="revolute" name="joint_front_right">
      <child>front_right</child>
      <parent>chassis</parent>
      <axis>
        <xyz>0 0 1</xyz>
        <dynamics>
          <friction>0.01</friction>
          <damping>0.01</damping>
        </dynamics>
      </axis>
    </joint>
    
    <link name="rear_left">
      <pose>-0.08 0.07 -0.01 1.5707 0 0</pose>
      <inertial>
        <mass>0.01</mass>
        <inertia>
          <ixx>0.0001</ixx>
          <iyy>0.0001</iyy>
          <izz>0.0001</izz>
        </inertia>
      </inertial>

      <collision name="collision">
        <geometry>
          <cylinder>
            <radius>0.03</radius>
            <length>0.02</length>
          </cylinder>
        </geometry>
      </collision>

      <visual name="tire">
        <geometry>
          <cylinder>
            <radius>0.03</radius>
            <length>0.02</length>
          </cylinder>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/DarkGrey</name>
          </script>
        </material>
      </visual>

      <visual name="wheel">
        <geometry>
          <cylinder>
            <radius>0.028</radius>
            <length>0.022</length>
          </cylinder>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/Yellow</name>
          </script>
        </material>
      </visual>

      <visual name="wheel_line">
        <geometry>
          <box>
            <size>0.053 0.01 0.024</size>
          </box>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/DarkGrey</name>
          </script>
        </material>
      </visual>

      <visual name="wheel_line2">
        <geometry>
          <box>
            <size>0.01 0.053 0.024</size>
          </box>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/DarkGrey</name>
          </script>
        </material>
      </visual>

    </link>

    <joint type="revolute" name="joint_rear_left">
      <child>rear_left</child>
      <parent>chassis</parent>
      <axis>
        <xyz>0 0 1</xyz>
        <dynamics>
          <friction>0.01</friction>
          <damping>0.01</damping>
        </dynamics>
      </axis>
    </joint>
    
    <link name="rear_right">
      <pose>-0.08 -0.07 -0.01 1.5707 0 0</pose>
      <inertial>
        <mass>0.01</mass>
        <inertia>
          <ixx>0.0001</ixx>
          <iyy>0.0001</iyy>
          <izz>0.0001</izz>
        </inertia>
      </inertial>

      <collision name="collision">
        <geometry>
          <cylinder>
            <radius>0.03</radius>
            <length>0.02</length>
          </cylinder>
        </geometry>
      </collision>

      <visual name="tire">
        <geometry>
          <cylinder>
            <radius>0.03</radius>
            <length>0.02</length>
          </cylinder>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/DarkGrey</name>
          </script>
        </material>
      </visual>

      <visual name="wheel">
        <geometry>
          <cylinder>
            <radius>0.028</radius>
            <length>0.022</length>
          </cylinder>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/Yellow</name>
          </script>
        </material>
      </visual>

      <visual name="wheel_line">
        <geometry>
          <box>
            <size>0.053 0.01 0.024</size>
          </box>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/DarkGrey</name>
          </script>
        </material>
      </visual>

      <visual name="wheel_line2">
        <geometry>
          <box>
            <size>0.01 0.053 0.024</size>
          </box>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/DarkGrey</name>
          </script>
        </material>
      </visual>

    </link>

    <joint type="revolute" name="joint_rear_right">
      <child>rear_right</child>
      <parent>chassis</parent>
      <axis>
        <xyz>0 0 1</xyz>
        <dynamics>
          <friction>0.01</friction>
          <damping>0.01</damping>
        </dynamics>
      </axis>
    </joint>
    

    <link name="ball">
      <pose>0.04 0 0.05 0 0 0</pose>
      <inertial>
        <mass>0.001</mass>
        <inertia>
          <ixx>0.0001</ixx>
          <iyy>0.0001</iyy>
          <izz>0.0001</izz>
        </inertia>
      </inertial>
      <visual name="ball_visual">
        <geometry>
          <sphere>
            <radius>0.03</radius>
          </sphere>
        </geometry>
        <material>
          <script>
            <uri>file://media/materials/scripts/gazebo.material</uri>
            <name>Gazebo/Red</name>
          </script>
        </material>
      </visual>
      <collision name="ball_collision">
        <geometry>
          <sphere>
            <radius>0.03</radius>
          </sphere>
        </geometry>
      </collision>
    </link>

    <joint type="fixed" name="ball_attachment">
      <child>ball</child>
      <parent>chassis</parent>
      <axis>
        <xyz>0 0 1</xyz>
        <dynamics>
          <friction>0.01</friction>
          <damping>0.01</damping>
        </dynamics>
      </axis>
    </joint>
 
  </model>
</sdf>

<!-- vim: set et fenc=utf-8 ff=unix sts=0 sw=2 ts=2 : -->